File: benchmarks/fibonacci/fib.lua
local function fib(n)
    if n < 2 then
        return n
    end
    return fib(n - 1) + fib(n - 2)
end

local n = 32
local result = fib(n)
print(string.format("fib(%d) = %d", n, result))

File: benchmarks/fibonacci/Fib.java
public class Fib {
    static int fib(int n) {
        if (n < 2) {
            return n;
        }
        return fib(n - 1) + fib(n - 2);
    }

    public static void main(String[] args) {
        int n = 35;
        int result = fib(n);
        System.out.printf("fib(%d) = %d%n", n, result);
    }
}
